python/lsst/daf/butler/core/formatter.py

```diff
@@ -50,11 +50,14 @@ class Formatter(metaclass=ABCMeta):
     are supported (`frozenset`).
     """
 
-    def __init__(self, fileDescriptor=None):
-        if fileDescriptor is not None and not isinstance(fileDescriptor, FileDescriptor):
+    def __init__(self, fileDescriptor):
+        if not isinstance(fileDescriptor, FileDescriptor):
             raise TypeError("File descriptor must be a FileDescriptor")
         self._fileDescriptor = fileDescriptor
 
+    def __str__(self):
+        return f"{self.name()}@{self.fileDescriptor.location.path}"
+
```

Comment:
Seems like it'd be straightforward to add the ideal kind of `__repr__`, too.

Reply:
Yes, it was straightforward here but required I added repr() to FileDescriptor and Location and also that I cleaned up repr() for StorageClass to make it less chatty.  You now get a repr that looks something like:
```
lsst.daf.butler.formatters.jsonFormatter.JsonFormatter(FileDescriptor(Location('/Users/timj/work/lsstsw3/build/daf_butler/a', 'B'), StorageClass('xxx'), readStorageClass=StorageClass('yyy')))
```
@TallJimbo you might want to take a look at all the new code now.

Reply:
Looks good.  I didn't know about `{!r}`, and I'm glad that I now do.